### ISSUE TITLE
What if there is no eyaml private key? 

### ISSUE THREAD

**Issue by timbirk** [ITV/domed-city#42]
https://github.com/ITV/domed-city/blob/master/lib/dome/hiera_lookup.rb#L36-L44

Makes the assumption that you have an eyaml key and throws a stacktrace:

```
bundler: failed to load command: dome (/Users/timbirk/.gem/ruby/2.3.1/bin/dome)
NoMethodError: undefined method `colorize' for nil:NilClass
  /Users/timbirk/.gem/ruby/2.3.1/bundler/gems/domed-city-26657b98ba2f/lib/dome/hiera_lookup.rb:42:in `eyaml_dir'
  /Users/timbirk/.gem/ruby/2.3.1/bundler/gems/domed-city-26657b98ba2f/lib/dome/hiera_lookup.rb:47:in `eyaml_private_key'
  /Users/timbirk/.gem/ruby/2.3.1/bundler/gems/domed-city-26657b98ba2f/lib/dome/hiera_lookup.rb:24:in `default_config'
  /Users/timbirk/.gem/ruby/2.3.1/bundler/gems/domed-city-26657b98ba2f/lib/dome/hiera_lookup.rb:13:in `config'
  /Users/timbirk/.gem/ruby/2.3.1/bundler/gems/domed-city-26657b98ba2f/lib/dome/hiera_lookup.rb:61:in `lookup'
  /Users/timbirk/.gem/ruby/2.3.1/bundler/gems/domed-city-26657b98ba2f/lib/dome/hiera_lookup.rb:73:in `block in secret_env_vars'
  /Users/timbirk/.gem/ruby/2.3.1/bundler/gems/domed-city-26657b98ba2f/lib/dome/hiera_lookup.rb:72:in `each_pair'
  /Users/timbirk/.gem/ruby/2.3.1/bundler/gems/domed-city-26657b98ba2f/lib/dome/hiera_lookup.rb:72:in `secret_env_vars'
  /Users/timbirk/.gem/ruby/2.3.1/bundler/gems/domed-city-26657b98ba2f/lib/dome/secrets.rb:15:in `secret_env_vars'
  /Users/timbirk/.gem/ruby/2.3.1/bundler/gems/domed-city-26657b98ba2f/lib/dome/terraform.rb:61:in `create_plan'
  /Users/timbirk/.gem/ruby/2.3.1/bundler/gems/domed-city-26657b98ba2f/lib/dome/terraform.rb:49:in `plan'
  /Users/timbirk/.gem/ruby/2.3.1/bundler/gems/domed-city-26657b98ba2f/bin/dome:30:in `<top (required)>'
  /Users/timbirk/.gem/ruby/2.3.1/bin/dome:23:in `load'
  /Users/timbirk/.gem/ruby/2.3.1/bin/dome:23:in `<top (required)>'
```

Which is not so useful.

**Comment by timbirk** [comment]
Furhter to the above this whole section looks a bit weird... https://github.com/ITV/domed-city/blob/master/lib/dome/hiera_lookup.rb#L36-L51

**Comment by timbirk** [comment]
Perhaps eyaml_dir would be better off named: eyaml_key_dir.

**Comment by kemra102** [comment]
@timbirk #41 add's an appropriate error message in just this situation

**Comment by dprint8** [comment]
Similar issue:

`bundler: failed to load command: dome (/Users/DimiPrin/.gem/ruby/2.3.1/bin/dome)
TypeError: no implicit conversion of nil into String
  /Users/DimiPrin/.gem/ruby/2.3.1/bundler/gems/domed-city-adfcde061943/lib/dome/hiera_lookup.rb:49:in `join'
  /Users/DimiPrin/.gem/ruby/2.3.1/bundler/gems/domed-city-adfcde061943/lib/dome/hiera_lookup.rb:49:in `eyaml_private_key'
  /Users/DimiPrin/.gem/ruby/2.3.1/bundler/gems/domed-city-adfcde061943/lib/dome/hiera_lookup.rb:24:in `default_config'
  /Users/DimiPrin/.gem/ruby/2.3.1/bundler/gems/domed-city-adfcde061943/lib/dome/hiera_lookup.rb:13:in `config'
  /Users/DimiPrin/.gem/ruby/2.3.1/bundler/gems/domed-city-adfcde061943/lib/dome/hiera_lookup.rb:67:in `lookup'
  /Users/DimiPrin/.gem/ruby/2.3.1/bundler/gems/domed-city-adfcde061943/lib/dome/hiera_lookup.rb:79:in `block in secret_env_vars'
  /Users/DimiPrin/.gem/ruby/2.3.1/bundler/gems/domed-city-adfcde061943/lib/dome/hiera_lookup.rb:78:in `each_pair'
  /Users/DimiPrin/.gem/ruby/2.3.1/bundler/gems/domed-city-adfcde061943/lib/dome/hiera_lookup.rb:78:in `secret_env_vars'
  /Users/DimiPrin/.gem/ruby/2.3.1/bundler/gems/domed-city-adfcde061943/lib/dome/secrets.rb:16:in `secret_env_vars'
  /Users/DimiPrin/.gem/ruby/2.3.1/bundler/gems/domed-city-adfcde061943/lib/dome/terraform.rb:34:in `plan'
  /Users/DimiPrin/.gem/ruby/2.3.1/bundler/gems/domed-city-adfcde061943/bin/dome:30:in `<top (required)>'
  /Users/DimiPrin/.gem/ruby/2.3.1/bin/dome:23:in `load'
  /Users/DimiPrin/.gem/ruby/2.3.1/bin/dome:23:in `<top (required)>'`

**Comment by kemra102** [comment]
The exact error reported by Tim above was fixed by #41 - suspect there are a few spots with similar assumptions.

**Comment by dprint8** [comment]
@themattkeating v6.2.0 as well
